sphinx/ext/apidoc.py

```diff
@@ -429,7 +429,7 @@ def main(argv: list[str] = sys.argv[1:]) -> int:
         ensuredir(args.destdir)
     excludes = tuple(
         re.compile(fnmatch.translate(path.abspath(exclude)))
-        for exclude in args.exclude_pattern
+        for exclude in dict.fromkeys(args.exclude_pattern)
```

Comment:
Out of curiosity, why `dict.fromkeys()` instead of `set()`?

Reply:
Probably overthinking this -- I wanted to de-duplicate patterns, but also e.g. you might want to order patterns in case there is a speed boost for more commonly excluded items. ``set`` would loose order, ``dict`` retains order. But this is fairly inconsequential!